test/unittests/libp2p/net.cpp

```diff
@@ -172,10 +199,10 @@ struct TestNodeTable: public NodeTable
     {
         NodeTable::onPacketReceived(_socket, _from, _packet);
 
-        packetReceived.set_value(_packet.toBytes());
```

Comment:
This was not working correctly when more than one packet was received - `set_value` throws if `promise` already has a value.